[package] bitserializer/>=0.50: broken headers when `with_rapidyaml=True`

### Description

When `with_rapidyaml` is set to True, the headers downloaded with prebuilt package are incorrect (not patched). In effect, when any code includes `bitserializer/rapidyaml_archive.h` then compiler returns error similar to:
```
$HOME/.conan2/p/bitse99e9d379ef622/p/include/bitserializer/rapidyaml_archive.h:15:10: fatal error: ryml/ryml_std.hpp: No such file or directory
```


### Package and Environment Details

* Package Name/Version: **bitserializer/0.70**
* Operating System+version: **Linux Ubuntu 22.04**
* Compiler+version: **GCC 11**
* Conan version: **conan 2.6.0**
* Python version: **Python 3.10.12**


### Conan profile

[settings]
arch=x86_64
os=Linux
compiler=gcc
compiler.version=11
#compiler=clang
#compiler.version=15
compiler.cppstd=20
#compiler.libcxx=libc++
compiler.libcxx=libstdc++11
build_type=Debug

[options]
[conf]
[build_requires]
[env]


### Steps to reproduce

```
conan install -r conancenter --requires=bitserializer/0.70 -o '&:with_rapidyaml=True'
head -n 20 $HOME/.conan2/p/bitse*/p/include/bitserializer/rapidyaml_archive.h
```
Notice the following lines:
```
#include <ryml/ryml_std.hpp>
#include <ryml/ryml.hpp>
```
which are unpatched includes (should have been patched for `with_rapidyaml=True`

### Logs

<details><summary>Click to expand log</summary>

```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64        
build_type=Debug               
compiler=gcc                          
compiler.cppstd=20
compiler.libcxx=libstdc++11               
compiler.version=11                                                            
os=Linux      
[options] 
&:with_rapidyaml=True
backward-cpp/*:stack_details=dw
backward-cpp/*:stack_walking=libunwind
[conf]            
tools.cmake.cmaketoolchain:generator=Ninja
                                                                               
Profile build:
[settings]
arch=x86_64                    
build_type=Debug                      
compiler=gcc
compiler.cppstd=20                        
compiler.libcxx=libstdc++11                                                    
compiler.version=11                                                            
os=Linux                                    
[options]                                                          
backward-cpp/*:stack_details=dw                 
backward-cpp/*:stack_walking=libunwind                                         
[conf]    
tools.cmake.cmaketoolchain:generator=Ninja
                                                                               
                                                                                                                                                               
======== Computing dependency graph ========              
bitserializer/0.70: Not found in local cache, looking in remotes...
bitserializer/0.70: Checking remote: conancenter            
bitserializer/0.70: Downloaded recipe revision b7090443b162e27f308a2d3633e55e09
Graph root                                    
    cli     
Requirements                                                                                                                                                   
    bitserializer/0.70#b7090443b162e27f308a2d3633e55e09 - Downloaded (conancenter)                                                  
    c4core/0.1.11#f4ab4857c7ade25213739379104e6717 - Cache                                                                                                     
    fast_float/6.1.0#1a10b6b2d8ffb3f69a7c72f2260496a4 - Cache                                                                                                  
    rapidyaml/0.5.0#fa55fd14369649067e0daa2c295aea76 - Cache                   
                                                                               
======== Computing necessary packages ========                                 
Requirements                           
    bitserializer/0.70#b7090443b162e27f308a2d3633e55e09:da39a3ee5e6b4b0d3255bfef95601890afd80709#dbc6976c356c50e6bfc4ad379669d57c - Download (conancenter)
    c4core/0.1.11#f4ab4857c7ade25213739379104e6717:39631168038a4d8cb5ef4ec62506619ae306ad77#3db295fa9bfbf63bc836e014e2ed3e64 - Cache                           
    fast_float/6.1.0#1a10b6b2d8ffb3f69a7c72f2260496a4:da39a3ee5e6b4b0d3255bfef95601890afd80709#38ce0a19014c0ff016772731a38dc33b - Cache
    rapidyaml/0.5.0#fa55fd14369649067e0daa2c295aea76:d3086d93dd282272a292b2fb07b9443fa5ea4aa7#06cfd44042f33a5fd51ccb45f8995cb2 - Cache                         
                                                                               
======== Installing packages ========       
                                                                                                                                                               
-------- Downloading 1 package --------                                                                                                                        
bitserializer/0.70: Retrieving package da39a3ee5e6b4b0d3255bfef95601890afd80709 from remote 'conancenter'                                                      
bitserializer/0.70: Package installed da39a3ee5e6b4b0d3255bfef95601890afd80709 
bitserializer/0.70: Downloaded package revision dbc6976c356c50e6bfc4ad379669d57c                                                                               
fast_float/6.1.0: Already installed! (1 of 4)
c4core/0.1.11: Already installed! (2 of 4)                           
rapidyaml/0.5.0: Already installed! (3 of 4)
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:                                                                    
WARN: deprecated:     'cpp_info.names' used in: rapidyaml/0.5.0, fast_float/6.1.0, bitserializer/0.70
WARN: deprecated:     'cpp_info.filenames' used in: bitserializer/0.70         
                                                                                                                                                                                                                                                                                                                              
======== Finalizing install (deploy, generators) ========                      
cli: Generating aggregated env files                                                                                                                                                                                                                                                                                          
cli: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
Install finished successfully  

/*******************************************************************************
* Copyright (C) 2020-2024 by Artsiom Marozau, Pavel Kisliak                    *
* This file is part of BitSerializer library, licensed under the MIT license.  *
*******************************************************************************/
#pragma once
#include <cassert>
#include <type_traits>
#include <optional>
#include <variant>
#include "bitserializer/serialization_detail/errors_handling.h"
#include "bitserializer/serialization_detail/archive_base.h"

// External dependency (Rapid YAML)
#include <c4/format.hpp>
#include <ryml/ryml_std.hpp>
#include <ryml/ryml.hpp>

namespace c4::yml
{
        // Dummy

```

</details>

--- a/recipes/bitserializer/all/conanfile.py
+++ b/recipes/bitserializer/all/conanfile.py
@@ -131,7 +131,7 @@ class BitserializerConan(ConanFile):
             deps.generate()
 
     def _patch_sources(self):
-        if Version(self.version) >= "0.50" and self.options.with_rapidyaml:
+        if Version(self.version) >= "0.50":
             # Remove 'ryml' subdirectory from #include
             replace_in_file(
                 self, os.path.join(self.source_folder, "include", "bitserializer", "rapidyaml_archive.h"),

